codec-http/src/main/java/io/netty/handler/codec/http/HttpContentEncoder.java

```diff
@@ -319,4 +327,22 @@ public EmbeddedChannel contentEncoder() {
             return contentEncoder;
         }
     }
+
+    private static final class RequestMetaData {
```

Comment:
Alternatively, you could define a singleton `CharSequence` instance which tells `HttpContentEncoder` to skip encoding completely.  Your approach is nice, but we can save object allocation with this trick.